blocks/search-results/search-results.js

```diff
@@ -0,0 +1,128 @@
+    createCards(resultData, resultsDiv);
+  }
+}
+
+function filterMatches(tokenizedSearchWords, jsonData) {
+  let allData = [];
+  tokenizedSearchWords.forEach((element) => {
+    const match = jsonData.data.filter((entry) => (
+      entry.title
+      + entry.content
+      + entry.path
+      + entry.authors
+      + entry.collections
+      + entry.section
+      + entry.categories
+    )
+      .toLowerCase()
+      .includes(element.toLowerCase()));
+    if (match) { allData = allData.concat(match); }
+  });
+  return uniqueMatches(allData);
+}
+
+function uniqueMatches(allData) {
+  const uniquePath = new Set();
+  return allData.filter((element) => {
+    // console.log(element.path);
+    const isDuplicate = uniquePath.has(element.path);
+    if (!isDuplicate) {
+      uniquePath.add(element.path);
+      return true;
+    }
+    return false;
+  });
+}
+
+function noResults(resultsDiv) {
+  document.querySelector('main .results-loading-spinner').style.display = 'none';
+  const sorryDiv = document.createElement('div');
+  const sorryPara = document.createElement('p');
+  sorryDiv.append(sorryPara);
+  sorryPara.textContent = 'Sorry, no results were found, search again ?';
+  resultsDiv.append(sorryDiv);
+  sorryDiv.classList.add('no-results');
+  const searchFormDiv = document.createElement('div');
+  sorryDiv.append(searchFormDiv);
+  searchFormDiv.innerHTML = `
+   <div class="search-container">
+    <div class="search-wrapper">
+     <div class='search-form'>
+      <form action='/search' method='get'>
+        <input type='search' name='q' class='search-input' placeholder="TYPE HERE"/>
+      </form>
+     </div>
+    </div>
+   </div>
+ `;
+  document.querySelector('.block.search-results.card-container.three-columns .results-div').style.display = 'block';
+}
+
+function displayNextEntries(iterator, numInitialLoadedArticles, loadMoreContainer, resultsDiv) {
+  if (done) { resultsDiv.parentNode.querySelector('.results-loading-spinner').remove(); done = false; }
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i <= numInitialLoadedArticles; i++) {
+    if (i === numInitialLoadedArticles) { resultsDiv.after(loadMoreContainer); }
+    const next = iterator.next();
+    if (next.done) {
+      loadMoreContainer.remove();
+    }
+    const searchItem = next.value;
+    if (!searchItem) break;
+    const wrapper = document.createElement('div');
+    createCardBlock(searchItem, wrapper);
+    resultsDiv.append(wrapper);
+  }
+}
+
+function createCards(finalArray, resultsDiv) {
+  const numInitialLoadedArticles = 23;
+  const iterator = finalArray.values();
+  const loadMoreContainer = document.createElement('div');
+  loadMoreContainer.classList.add('article-load-more-container');
+  loadMoreContainer.innerHTML = '<button class="article-list-load-more-button">Load more</button>';
+  displayNextEntries(iterator, numInitialLoadedArticles, loadMoreContainer, resultsDiv);
+  loadMoreContainer.addEventListener('click', () => {
```

Comment:
I would listen on the button, not the container. 
```suggestion
  loadMoreContainer.querySelector('button').addEventListener('click', () => {
```

Reply:
Fixed